### PR TITLE
Mask API key for AI21 LLM

### PR DESCRIPTION
**Pull request by gautamanirudh**
- **Description:** Added masking of the API Key for AI21 LLM when printed and improved the docstring for AI21 LLM.
    - Updated the AI21 LLM to utilize SecretStr from pydantic to securely manage API key.
    - Made improvements in the docstring of AI21 LLM. It now mentions that the API key can also be passed as a named parameter to the constructor.
    - Added unit tests.
  - **Issue:** #12165 
  - **Tag maintainer:** @eyurtsev

### CODE REVIEW

**Comment by gautamanirudh** [comment]
@eyurtsev Does it make sense to move the `_to_secret()` function(first implemented in #12377) to langchain utils? It can be utilized in other files as well.

**Comment by eyurtsev** [comment]
@gautamanirudh yeah let's move to utils

File: libs/langchain/langchain/llms/ai21.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ class AI21(LLM):
     logitBias: Optional[Dict[str, float]] = None
     """Adjust the probability of specific tokens being generated."""
 
-    ai21_api_key: Optional[str] = None
+    ai21_api_key: Optional[SecretStr] = None
```
Comment (eyurtsev) [inline libs/langchain/langchain/llms/ai21.py:65]: Is it possible for the ai21_api_key to be None? It looks like after the root validator runs this always get set to a value?

File: libs/langchain/langchain/llms/ai21.py [hunk 2]
```diff
@@ -141,9 +143,10 @@ def _call(
             else:
                 base_url = "https://api.ai21.com/studio/v1"
         params = {**self._default_params, **kwargs}
+        self.ai21_api_key = cast(SecretStr, self.ai21_api_key)
```
Comment (eyurtsev) [inline libs/langchain/langchain/llms/ai21.py:146]: Cast probably won't be needed if the attribute is no longer an Optional

File: libs/langchain/tests/unit_tests/llms/test_ai21.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+"""Test AI21 llm"""
```
Comment (eyurtsev) [inline libs/langchain/tests/unit_tests/llms/test_ai21.py:1]: Excellent!

**Comment by eyurtsev** [comment]
Ready to merge -- we might want to update the type to be non optional later on, but not requiring changes for this PR since ti was already optional before

**Review by eyurtsev — APPROVED** [review]

File: libs/langchain/langchain/llms/ai21.py (hunk 1 above)
Comment (gautamanirudh) [reply libs/langchain/langchain/llms/ai21.py:65]: Yes, ai21_api_key can never be None.

File: libs/langchain/langchain/llms/ai21.py (hunk 2 above)
Comment (gautamanirudh) [reply libs/langchain/langchain/llms/ai21.py:146]: Yes, used the cast to avoid type checking, I was encountering a mypy error without it.

**Pull request merged**